ArkhamOverlay/Pages/LocalImages/LocalImagesController.cs

```diff
@@ -0,0 +1,204 @@
+
+            }
+
+            var packs = new List<LocalPack>();
+            try {
+                _logger.LogMessage($"Loading packs from {_appData.Configuration.LocalImagesDirectory}.");
+                foreach (var directory in Directory.GetDirectories(_appData.Configuration.LocalImagesDirectory)) {
+                    packs.Add(LoadPack(directory));                    
+                }
+                ViewModel.Packs = packs;
+            } catch (Exception e) {
+                _logger.LogError(e.Message);
+            }
+        }
+
+        private LocalPack LoadPack(string directory) {
+            var pack = new LocalPack(directory);
+
+            var manifestPath = directory + "\\Manifest.json";
+            if (File.Exists(manifestPath)) {
+                try {
+                    _logger.LogMessage($"Loading pack manifest {manifestPath}.");
+                    var manifest = JsonConvert.DeserializeObject<PackManifest>(File.ReadAllText(manifestPath));
+                    ReadManifest(manifest, pack);
+
+                    pack.PropertyChanged += (s, e) => {
+                        if (e.PropertyName == nameof(LocalPack.SelectedCard) || e.PropertyName == nameof(LocalPack.IsCardSelected)) {
+                            return;
+                        }
+
+                        WriteManifest();
+                    };
+                } catch (Exception e) {
+                    _logger.LogError(e.Message);
+                }
+            }
+
+            return pack;
+        }
+
+        private LocalCard LoadCard(LocalPack pack, string filePath) {
+            if (string.Compare(Path.GetExtension(filePath), ".json", StringComparison.InvariantCulture) == 0) {
+                return null;
+            }
+
+            //the backs of cards will be loaded with the front- adding -back to a file denotes it's the back of a card
+            if (Path.GetFileNameWithoutExtension(filePath).ToLower().EndsWith("-back")) {
+                return null;
+            }
+            
+            try {
+                var card = pack.Cards.FirstOrDefault(x => string.Compare(x.FilePath, filePath, StringComparison.InvariantCulture) == 0);
+                if (card == null) {
+                    card = new LocalCard(filePath);
+                    card.PropertyChanged += (s, e) => {
+                        WriteManifest();
+                    };
+                }
+
+                LoadCardImages(card);
+
+                return card;
+            } catch {
+                //this isn't an image- that's completely valid, we just ignore it and move on
+                return null;
+            }
+        }
+
+        private void LoadCardImages(LocalCard card) {
+            if (card.Image != null) {
+                return;
+            }
+
+            card.FrontThumbnail = ShellFile.FromFilePath(card.FilePath).Thumbnail.BitmapSource;
```

Comment:
I'd consider adding a try around this code. No idea if it can throw or not, but it probably can.

Reply:
We have a try around it that doesn't log- as it's the only way I know of to detect non image files.